ocaml/database/block_device_io.ml

```diff
@@ -476,11 +476,12 @@ let action_writedb block_dev_fd client datasock target_response_time =
     (* If there's space, write some ASCII NULs over the next few bytes so that we trample on any data which may already exist on the block device *)
     let remaining_space = remaining_space - min_space_needed in
     let trample_size = if size_size > remaining_space then remaining_space else size_size in
-    Unixext.time_limited_write block_dev_fd trample_size (String.make trample_size '\000') target_response_time;
+    Unixext.time_limited_write block_dev_fd trample_size (Bytes.make trample_size '\000') target_response_time;
 
     (* Seek backwards in the block device to where the length is supposed to go and write it *)
     ignore_int (Unixext.seek_to block_dev_fd pos_to_write_length);
-    Unixext.time_limited_write block_dev_fd size_size (Printf.sprintf "%016d" total_length) target_response_time;
+    let total_length_str = Printf.sprintf "%016d" total_length in
+    Unixext.time_limited_write block_dev_fd size_size (Bytes.of_string total_length_str) target_response_time;
```

Comment:
This could use Unixext.time_limited_write_substring

Reply:
Here it makes sense to use the substring one to avoid a copy, I'll change it